plugins/services/src/js/containers/service-connection/ServiceConnectionEndpointList.js

```diff
@@ -0,0 +1,244 @@
+import React from "react";
+import { routerShape } from "react-router";
+
+import ConfigurationMap from "#SRC/js/components/ConfigurationMap";
+import ConfigurationMapHeading
+  from "#SRC/js/components/ConfigurationMapHeading";
+import ConfigurationMapLabel from "#SRC/js/components/ConfigurationMapLabel";
+import ConfigurationMapRow from "#SRC/js/components/ConfigurationMapRow";
+import ConfigurationMapSection
+  from "#SRC/js/components/ConfigurationMapSection";
+import ConfigurationMapValue from "#SRC/js/components/ConfigurationMapValue";
+import ClipboardTrigger from "#SRC/js/components/ClipboardTrigger";
+import Icon from "#SRC/js/components/Icon";
+
+import ServiceNoEndpointsPanel from "./ServiceNoEndpointsPanel";
+import Service from "../../structs/Service";
+import ServiceConfigUtil from "../../utils/ServiceConfigUtil";
+import { getDisplayValue } from "../../utils/ServiceConfigDisplayUtil";
+
+const METHODS_TO_BIND = ["handleOpenEditConfigurationModal", "handleTextCopy"];
+
+class ServiceConnectionEndpointList extends React.Component {
+  constructor() {
+    super(...arguments);
+
+    this.state = {
+      copiedCommand: ""
+    };
+
+    METHODS_TO_BIND.forEach(method => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  handleTextCopy(copiedCommand) {
+    this.setState({ copiedCommand });
+  }
+
+  handleOpenEditConfigurationModal() {
+    const { router } = this.context;
+    router.push(
+      `/services/detail/${encodeURIComponent(this.props.service.getId())}/edit/`
+    );
+  }
+
+  getClipboardTrigger(command) {
```

Comment:
Not a blocker, but I'd prefer local or _private_ components instead of all these helper methods. 

/cc @dcos/frontend-foundation

Reply:
Extracted to a separate component

Reply:
Now that we have a `EndpointClipboardTrigger` I don't see any need for a `getClipboardTrigger` method. Anyway, feel free to address this in a later clean up PR.

Reply:
...only if there will be a clean up pr following. Otherwise let's do it right away please :)